template/native_enum_type_traits.njk

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2017 The Bacardi Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GEN_NATIVE_ENUM_TYPE_TRAITS_H_
+#define GEN_NATIVE_ENUM_TYPE_TRAITS_H_
+
+#include <napi.h>
+#include <string>
+#include <type_traits>
+#include "core/native_type_traits.h"
+#include "idl_enum_type.h"
+
+{% for enum in enums %}
+template <>
+struct NativeTypeTraits<IDL{{enum.name}}>
+    : public NativeTypeTraitsBase<IDL{{enum.name}}> {
+  static std::string NativeValue(const Napi::Env& env,
+                                 const Napi::Value& js_value) {
+    if (!js_value.IsString()) {
+      Napi::TypeError::New(env, "It's an invalid string.")
+          .ThrowAsJavaScriptException();
+      return std::string();
+    }
+
+    std::string value = js_value.ToString().Utf8Value();
+    if (!IsValidValue(value)) {
+      Napi::TypeError::New(env, "it not matched with values of enum in idl.")
+          .ThrowAsJavaScriptException();
+      return std::string();
+    }
+
+    return js_value.ToString().Utf8Value();
+  }
+
+  static bool IsTypeEquals(const Napi::Value& js_value) {
```

Comment:
This is an another problem. IsTypeEquals() is used for method overloading check, right?
If developers defines two enums and then two method overloading for each enums, unexpected behavior might occur.
```
enum A {
  ...
};

enum B {
  ...
};

void method(A a);
void method(B b);
```
In WebIDL level, they seems to have different signature but the enum is just string in JS level. So, they can not be method-overloaded. So, it's a reason that we should not generate this file repeatedly.